File: nodejs/gmsec5/dox/U64Field.java
/*
 * Copyright 2007-2023 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


/**
 * @file U64Field.java
 */

package gmsec;


/**
 * Specialized Field class that can be used to retain an unsigned 64-bit value.
 */
public class U64Field extends Field
{
	/**
	 * Constructor.
	 *
	 * @param name Name of the field.
	 * @param data An unsigned 64-bit value.
	 *
	 * @throws SWIG_Exception Thrown if the field name is null.
	 * @throws GmsecException Thrown if the field name contains an empty string.
	 */
	public U64Field(name, data);


	/**
	 * Constructor.
	 *
	 * @param name      Name of the field.
	 * @param data      An unsigned 64-bit value.
	 * @param is_header Used to indicate if Field is a header field.
	 *
	 * @throws SWIG_Exception Thrown if the field name is null.
	 * @throws GmsecException Thrown if the field name contains an empty string.
	 */
	public U64Field(name, data, is_header);


	/**
	 * Copy constructor.
	 *
	 * @param other The other U64Field object to copy.
	 *
	 * @throws SWIG_Exception Thrown if the given U64Field is null.
	 */
	public U64Field(other);


	/**
	 * Returns the data stored within the U64Field.
	 *
	 * @return An unsigned 64-bit value.
	 */
	public getValue();
}

File: ruby/gmsec5/dox/U64Field.rb
# Copyright 2007-2024 United States Government as represented by the
# Administrator of The National Aeronautics and Space Administration.
# No copyright is claimed in the United States under Title 17, U.S. Code.
# All Rights Reserved.

# This file is used for generating Ruby documentation; it only contains
# stubs of constructors and methods associated with the class.
#


module Libgmsec_ruby #:nodoc: don't document this

	# This is a specialized Field class that can be used to store a 64-bit unsigned integer value.
	#
	# ==== Example Usage
	#
	#    require 'libgmsec_ruby'
	#
	#    begin
	#        field = Libgmsec_ruby::U64Field.new("MY-U64-FIELD", 3)
	#    end
	#
	class U64Field < Libgmsec_ruby::Field
		# Constructor.
		#
		# ==== Attributes
		#
	    # * +name+  - The name of the field.
	    # * +value+ - The value of the field.
        # * +is_header+ - Used to indicate if Field is a header field (default is false).
		#
		# ==== Raises
		#
		# A GmsecException is thrown if the name is nil, is an empty string, or is otherwise non-compliant.
		#
		def initialize(name, value, is_header)
		end


		# Returns the value associated with the field.
		#
		# ==== Returns
		#
		# A 64-bit unsigned integer value.
		#
		def get_value
		end
	end
end
